tt_metal/impl/device/program_cache.hpp

```diff
@@ -18,7 +18,7 @@ namespace detail {
 // Generic Program Cache: This data structure is tied to a device handle and can store generic program types from
 // TT-Metal and TT-Eager using tt::stl::concepts::unique_any.
 struct ProgramCache {
-    inline bool contains(uint64_t program_hash) { return this->cache_.count(program_hash) > 0; }
+    inline bool contains(uint64_t program_hash) { return is_enabled() ? this->cache_.count(program_hash) > 0 : false; }
 
     template <typename T>
     inline T& get(uint64_t program_hash) {
```

Comment:
Did you consider disabling this too?

Reply:
Have no idea what to return? TT_FATAL then?